chipy_org/apps/job_board/models.py

```diff
@@ -0,0 +1,95 @@
+                   ('approved', 'Approved'),
+                   ('denied', 'Denied'),
+                   ('extended', 'Extended'),
+                   ('archived', 'Archived') 
+                 ]
+
+
+class JobPost(CommonModel):
+    
+    __original_status = None
+
+    company_name = models.CharField(max_length=MAX_LENGTH)
+    
+    position = models.CharField(max_length=MAX_LENGTH)
+    
+    description = models.CharField(max_length=2500, help_text = '2500 Character Limit')
+
+    is_sponsor = models.BooleanField(
+        default=False, verbose_name="Is the company a sponsor of ChiPy?")
+
+    company_sponsor = models.ForeignKey(Sponsor, blank=True, null=True, verbose_name="If your company is a sponsor of ChiPy, please select it from this list.")
+
+    # After checking to see that the company_name and company_sponsor match, it is then considered verified
+    is_verified_sponsor = models.BooleanField(editable=False, default=False)
+
+    can_host_meeting = models.BooleanField(
+        default=False, verbose_name="Is your organization interested in hosting an event?")
+
+    status = models.CharField(max_length=20, choices=STATUS_CHOICES , default='pending')
+    
+    # - 'status_change_date' is used to calculate relative time, i.e. 3 Days from when Job was first approved for posting
+    # - Will also be useful for calculating when post should expire 
+    # - This field will change every time you update the 'status' field. Code for doing that is in the 'save' method.
+    status_change_date = models.DateTimeField(editable=False, auto_now_add=True) 
+
+    link_to_company_page = models.CharField(max_length=MAX_LENGTH)
+
+    contact = models.ForeignKey(User, blank=True, null=True)
+
+    agree_to_terms = models.BooleanField(
+        verbose_name="Does the user agree to referral terms?")
+
+    def __str__(self):
+        return f"{self.position} at {self.company_name}"
+
+    def __init__(self, *args, **kwargs):
+        super(JobPost, self).__init__(*args, **kwargs)
+        self.__original_status = self. status
+
+    def save(self, *args, **kwargs):
+        # Every time a decision is made for the status of the post, 
+        # the date that the decision is made is updated. 
+        
+        if  self.__original_status != self.status:
+            self.status_change_date = datetime.now()
+            self.__original_status = self. status
+ 
+        # The property 'verify_sponsor' is saved in a model field because
+        # properties can not be used in filtering querysets.
+        # Query filtering and ordering in Django views are based only on fields, not properties.
+        self.is_verified_sponsor = self.verify_sponsor
+
+        return super(JobPost, self).save(*args, **kwargs)
+
+    @property
+    def days_elapsed(self):
+
+        # computes days elapsed from when job post is put in 'approved' or 'extended' status
+        if self.status == 'approved' or self.status == 'extended':
```

Comment:
Yes, I've removed 'extended'. I also created an 'expiration_date' field, as well as a 'days_to_expire' field. I explain what these are in the pictures below that I posted after I made another commit.

Reply:
Let's not worry about expired jobs just yet. I think we're losing track that there are currently 0 jobs on our jobs board and we're still figuring out what needs to get built.

My experience has taught me that you should get the most basic MVP out the door and into the hands of customers / users. Only when people use your software, you understand what you need to build.

As long as we have a `created_at` and `modified_at` field, we have up to 60 days since this goes live to write up the expiration feature.

Reply:
@alysivji Thanks for your feed back. I have coded in the expired feature already. It currently defaults to 30 days, but I can change it to 60 days in my next commit, which probably will happen sometime soon. I'm going to keep this feature in there if that's okay with you. If for some reason you don't like the way it's working, we can revisit it in the future and I can rework it. 

Also, I asked @eevelweezel to take a look at my PR and depending on her comments, I might make more changes.

Reply:
I like the idea of allowing this to be user-editable, and keeping it as simple as possible. This functionality sounds reasonable, for the time being.